roles/ipaclient/library/ipaclient_setup_nss.py

```diff
@@ -430,13 +436,31 @@ def main():
             # Modify nsswitch/pam stack
             # pylint: disable=deprecated-method
             argspec = getargspec(tasks.modify_nsswitch_pam_stack)
-            if "sudo" in argspec.args:
+            if "sudo" in argspec.args and 'subid' in argspec.args:
+                tasks.modify_nsswitch_pam_stack(
+                    sssd=options.sssd,
+                    mkhomedir=options.mkhomedir,
+                    statestore=statestore,
+                    sudo=options.conf_sudo,
+                    subid=options.subid
+                )
+            elif 'sudo' in argspec.args and 'subid' not in argspec.args:
                 tasks.modify_nsswitch_pam_stack(
                     sssd=options.sssd,
                     mkhomedir=options.mkhomedir,
                     statestore=statestore,
                     sudo=options.conf_sudo
                 )
+            # The condition below is probably never used. There is no
+            # "ipa-client-install" version with `subid` and without
+            # `no-sudo` option
+            elif 'subid' in argspec.args and 'sudo' not in argspec.args:
+                tasks.modify_nsswitch_pam_stack(
+                    sssd=options.sssd,
+                    mkhomedir=options.mkhomedir,
+                    statestore=statestore,
+                    subid=options.subid
+                )
```

Comment:
Wouldn't this code be simpler if we only add the relevant option to a dict and then call `tasks.modify_nssswitch_pam_stack(**the_options)`?

Something like:
```python
the_options = {
    "sssd": options.sssd,
    "mkhomedir": options.mkhomedir,
    "statestore": statestore,
}
if "sudo in argspec.args:
   the_options["sudo"] = options.conf_sudo
if "subid" in argspec.args:
  the_options["subid"] = options.subid

tasks.modify_nss_switch_pam_stack(**the_options)
```

Reply:
Thanks @rjeffman, great suggestion!
I rechecked this patch against the downstream test in the repo `ansible-freeipa-tests` and it passed. See MR#473 -> latest pipeline -> TestClientTC28.test_nss_setup.